x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/results/LegacyTextEmbeddingResults.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.results;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.inference.InferenceResults;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Writes a text embedding result in the following json format
+ * {
+ *     "text_embedding": [
+ *         {
+ *             "embedding": [
+ *                 0.1
+ *             ]
+ *         },
+ *         {
+ *             "embedding": [
+ *                 0.2
+ *             ]
+ *         }
+ *     ]
+ * }
+ * @deprecated use {@link TextEmbeddingResults} instead
+ */
+@Deprecated
+public record LegacyTextEmbeddingResults(List<Embedding> embeddings) implements InferenceResults {
```

Comment:
I think we could fold this into the `TextEmbeddingResults` class by simply making it implement `InferenceResults` as well. I wasn't sure if that was less clear though or how that'd impact adding optional fields in the future. Or how it'd impact us if the `InferenceResults` interface changes in the future. I'm open to other ideas though.

Reply:
It's good to keep this as a separate class rather than cluttering up `TextEmbeddingResults`